internal/error.go

```diff
@@ -196,6 +196,18 @@ type (
 		// VersioningIntent specifies whether the continued workflow should run on a worker with a
 		// compatible build ID or not. See VersioningIntent.
 		VersioningIntent VersioningIntent
+
+		// This is optional and may be used to override the retry policy which gets carried over to the next run.
+		// If not set, the existing retry policy will be carried over automatically.
+		// For backward compatibility, we can't introduce an option, say WithWorkflowRetryPolicy,
+		// to override the retry policy attached to the workflow context.
+		// See #676 for more details.
+		RetryPolicy *commonpb.RetryPolicy
```

Comment:
```suggestion
		RetryPolicy *RetryPolicy
```

Can we make this the user-friendly version?

Reply:
> WorkflowOptions

Which workflow options? `client.StartWorkflowOptions` accepts a non-proto `RetryPolicy` as does `workflow.ChildWorkflowOptions`, but it's possible we have left some proto ones around.

> ExecuteActivityOptions

Which execute activity options? `workflow.ActivityOptions` accepts a non-proto `RetryPolicy`, but it's possible we have left some proto ones around.

Reply:
Not all of those are user facing and it's acknowledged that we put raw input/headers. But I think most of our user-facing options sets that accept retry policy (start workflow options, child workflow options, activity options, etc) accept the non-proto form.

Reply:
> If it can be used interchangeably, why do we need commonpb.RetryPolicy in the first place?

Explained in Slack, but that's the proto our API recognizes but is a bit annoying for users to use so we have a higher level form.